<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1126</width>
    <height>775</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QListView" name="listView">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>-20</y>
      <width>1101</width>
      <height>811</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>11</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>20</y>
      <width>401</width>
      <height>91</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>48</pointsize>
     </font>
    </property>
    <property name="text">
     <string>问题求解</string>
    </property>
   </widget>
   <widget class="QPushButton" name="file_choose">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>150</y>
      <width>141</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>选择文件</string>
    </property>
   </widget>
   <widget class="QLabel" name="saved_path_label">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>300</y>
      <width>331</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>解析成txt文件路径（供验证）：</string>
    </property>
   </widget>
   <widget class="QLabel" name="judge_label">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>650</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>结果判定为：</string>
    </property>
   </widget>
   <widget class="QLabel" name="judge">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>640</y>
      <width>141</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>未求解</string>
    </property>
   </widget>
   <widget class="QLabel" name="dpll_address_label">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>470</y>
      <width>191</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>求解cnf文件地址：</string>
    </property>
   </widget>
   <widget class="QLabel" name="cdcl_time_label">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>530</y>
      <width>191</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>优化算法实现时间：</string>
    </property>
   </widget>
   <widget class="QLabel" name="cdcl_time">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>540</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>（未实现）</string>
    </property>
   </widget>
   <widget class="QLabel" name="cdcl_address_label">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>570</y>
      <width>181</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>求解cnf文件地址：</string>
    </property>
   </widget>
   <widget class="QLabel" name="rate_label">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>640</y>
      <width>131</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>优化率：</string>
    </property>
   </widget>
   <widget class="QLabel" name="optimization_rate">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>650</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="text">
     <string>未完全求解</string>
    </property>
   </widget>
   <widget class="QPushButton" name="upload_back">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>30</y>
      <width>131</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="text">
     <string>返回</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>150</y>
      <width>311</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>请输入文件路径（cnf格式）：</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="cnf_path">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>230</y>
      <width>921</width>
      <height>31</height>
     </rect>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="save_button">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>290</y>
      <width>221</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>打开解析文件</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="saved_path">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>360</y>
      <width>931</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="source_button">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>290</y>
      <width>211</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>打开原文件</string>
    </property>
   </widget>
   <widget class="QPushButton" name="solve_button">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>140</y>
      <width>151</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>基础求解</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="dpll_address">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>470</y>
      <width>771</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="cdcl_address">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>590</y>
      <width>771</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="dpll_button">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>420</y>
      <width>331</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>打开基础算法求解文件</string>
    </property>
   </widget>
   <widget class="QPushButton" name="cdcl_button">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>540</y>
      <width>331</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>打开优化算法求解文件</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>410</y>
      <width>351</width>
      <height>51</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="dpll_time_label">
       <property name="font">
        <font>
         <family>Arial</family>
         <pointsize>14</pointsize>
        </font>
       </property>
       <property name="text">
        <string>基础算法实现时间：</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="dpll_time">
       <property name="font">
        <font>
         <family>Arial</family>
         <pointsize>14</pointsize>
        </font>
       </property>
       <property name="text">
        <string>（未实现）</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="better_solve_button_1">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>130</y>
      <width>111</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="text">
     <string>优化1</string>
    </property>
   </widget>
   <widget class="QPushButton" name="better_solve_button_2">
    <property name="geometry">
     <rect>
      <x>840</x>
      <y>130</y>
      <width>111</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="text">
     <string>优化2</string>
    </property>
   </widget>
   <widget class="QPushButton" name="better_solve_button_3">
    <property name="geometry">
     <rect>
      <x>960</x>
      <y>130</y>
      <width>111</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="text">
     <string>优化3</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1126</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>file_choose</sender>
   <signal>clicked()</signal>
   <receiver>cnf_path</receiver>
   <slot>show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>540</x>
     <y>178</y>
    </hint>
    <hint type="destinationlabel">
     <x>621</x>
     <y>221</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
